src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -0,0 +1,192 @@
+        bart = Persistence.feedSources.getById(bart.id);
+        bart.transformRules = new ArrayList<>();
+        Persistence.feedSources.replace(bart.id, bart);
+    }
+
+    /**
+     * Run tear down after each test. This just deletes the feed versions that were used in the test.
+     */
+    @After
+    public void tearDownTest() {
+        // Clean up
+        if (bartVersion != null) bartVersion.delete();
+        if (newBartVersion != null) newBartVersion.delete();
+    }
+
+    @Test
+    public void canReplaceGtfsPlusFile() throws IOException {
+        final String table = "stop_attributes";
+        bartVersion = createFeedVersionFromGtfsZip(bart, "bart_old.zip");
+        // Replace file transformation runs before feed is loaded into database.
+        // Note: stop_attributes.txt is a GTFS+ file found in BART's feed.
+        FeedTransformation transformation = ReplaceFileTransformation.create(bartVersion.id, table);
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        bart.transformRules.add(transformRules);
+        Persistence.feedSources.replace(bart.id, bart);
+        // Create new BART version (note: bart_new.zip GTFS file has been stripped of stop_attributes.txt)
+        newBartVersion = createFeedVersionFromGtfsZip(bart, "bart_new.zip");
+        // Check that new version has stop_attributes file
+        ZipFile zip = new ZipFile(newBartVersion.retrieveGtfsFile());
+        ZipEntry entry = zip.getEntry(table + ".txt");
+        assertThat(entry, Matchers.notNullValue());
+        // TODO Verify that stop_attributes file matches source file exactly?
+    }
+
+    @Test
+    public void canDeleteTrips() {
+        // Add delete trips transformation.
+        List<String> values = new ArrayList<>();
+        // Collect route_id values.
+        values.add("19");
+        FeedTransformation transformation = DeleteRecordsTransformation.create(
+            "trips",
+            "route_id",
+            values
+        );
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        bart.transformRules.add(transformRules);
+        Persistence.feedSources.replace(bart.id, bart);
+        // Load feed.
+        bartVersion = createFeedVersionFromGtfsZip(bart, "bart_old.zip");
+        // Fetch snapshot where modifications were made and create new version from it.
+        Snapshot snapshotWithModifications = bart.retrieveSnapshots().iterator().next();
+        CreateFeedVersionFromSnapshotJob newVersionJob = new CreateFeedVersionFromSnapshotJob(bart, snapshotWithModifications, user);
+        newVersionJob.run();
+        // Grab the modified version and check that the trips count matches expectation.
+        FeedVersion newVersion = bart.retrieveLatest();
+        assertEquals(
+            "trips count for transformed feed should be decreased by the # of records matched by the query",
+            bartVersion.feedLoadResult.trips.rowCount - 1041,
```

Comment:
Using the BART GTFS for these tests is somewhat problematic here. For starters, the BART GTFS file contains 33k records in the stop_times.txt, so it will take longer to run than a GTFS feed with the minimal amount of data needed to demonstrate test case functionality. Next up, with so much data it is hard to verify stuff like this. Before making this magic number, did we look at all 2,512 trips in the feed to see if exactly 1,041 less should be present? Finally, it's not easy to view changes in version control to a zip file if it has to change.

I feel these tests would be much better to reason about if they all used a GTFS feed obtained from a new `fake-agency-*` folder that has close to the minimum amount of data required to demonstrate applicable test case functionality.